.github/workflows/ci.yml

```diff
@@ -0,0 +1,324 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - release-*
+  pull_request: {}
+  workflow_dispatch: {}
+
+env:
+  # Common versions
+  GO_VERSION: '1.16'
+  GOLANGCI_VERSION: 'v1.31'
+  DOCKER_BUILDX_VERSION: 'v0.4.2'
```

Comment:
v0.4.2 is 1+ year old, should we update buildx? The current version is 0.7.1, My Docker Desktop on a Mac M1 is showing v0.6.3.

Reply:
Let's do that in a follow up. I'd prefer to keep this PR mostly scoped to copying the config we're using everywhere else. (Keeping that config updated across all our repos is another problem.)